halpybot/commands/caseutils.py

```diff
@@ -7,21 +7,60 @@
 Licensed under the GNU General Public License
 See license.md
 """
-from typing import List, Union
+import functools
+import re
+from typing import List
 from pendulum import now
+from halpybot import config
 from ..packages.command import Commands, get_help_text
-from ..packages.models import Context, User, NoUserFound, Case
+from ..packages.edsm import (
+    sys_cleaner,
+    checklandmarks,
+    NoResultsEDSM,
+    EDSMConnectionError,
+)
+from ..packages.models import (
+    Context,
+    User,
+    NoUserFound,
+    Case,
+    Status,
+    CaseType,
+    Platform,
+    KFCoords,
+    KFType,
+)
+from ..packages.checks import Require, Drilled
+from ..packages.case import get_case, update_single_elem_case_prep
 
 
-async def get_case(ctx: Context, case_arg: str) -> Case:
-    """Fetch a case from the Board"""
-    try:
-        caseref: Union[str, int] = int(case_arg)
-    except ValueError:
-        caseref = case_arg
-    return ctx.bot.board.return_rescue(caseref)
+# Decorators
+class CaseUtils:
+    """Utilities for Wrapping EDSM Commands"""
+
+    @staticmethod
+    def gather_case(len_args_expected: int):
```

Comment:
this `len_args_expected` thing feels bugprone, I am not convinced it should even be a thing.